scanpy/preprocessing/_pca.py

```diff
@@ -151,19 +157,52 @@ def pca(
 
     logg.info(f'    with n_comps={n_comps}')
 
-    random_state = check_random_state(random_state)
-
     X = adata_comp.X
 
-    if chunked:
+    is_dask = isinstance(X, DaskArray)
+
+    random_state = check_random_state(random_state) if not is_dask else random_state
+
+    if is_dask and not chunked:
+        # we will only let the user choose the randomized solver otherwise
+        # we will use our default
+        if svd_solver != 'randomized' and svd_solver != 'auto':
+            logg.debug('Ignoring svd_solver')
```

Comment:
Good idea to log it! Please add to the parameter documentation which values are valid when.

Maybe the default should be changed to `'auto'` or `None` if the actual default is different depending on the type of X?

Reply:
I would agree, but since it was auto before, I thought there should've been a good reason for them to change it. But I can also change it again if you think it's a good idea

Reply:
- Our default value shouldn’t trigger a warning (even if it’s just a log message here). It’s OK to translate our default value to another value for some or all implementations
- It should be possible for the user to specify all possible values that implementations accept.

E.g. if `'auto'` is accepted by some implementation, but we want that implementation to be called with `svd_solver='foo'` by default, our default should be e.g. `svd_solver=None` which then causes us to pass `svd_solver='foo'` to the implementation, but if an user sets `svd_solver='auto'`, that should be passed directly to any implementation that supports it.